File: 1769 - Minimum Number of Operations to Move All Balls to Each Box/Minimum Number of Operations to Move All Balls to Each Box.py
class Solution:
    def minOperations(self, boxes: str) -> List[int]:
        result = [0] * len(boxes)
        ops, count = 0, 0
        for i in range(len(boxes)):
            result[i] += ops
            count += 1 if boxes[i] == '1' else 0
            ops += count
        ops, count = 0, 0
        for i in range(len(boxes) - 1, -1, -1):
            result[i] += ops
            count += 1 if boxes[i] == '1' else 0
            ops += count
        return result

File: 1769 - Minimum Number of Operations to Move All Balls to Each Box/Minimum Number of Operations to Move All Balls to Each Box.cpp
class Solution {
public:
    vector<int> minOperations(const string& boxes) {
        vector<int> result(boxes.size());
        for (int i = 0, cnt = 0, ops = 0; i < boxes.size(); ++i) {
            result[i] += ops;
            cnt += boxes[i] == '1' ? 1 : 0;
            ops += cnt;
        }
        for (int i = boxes.size() - 1, cnt = 0, ops = 0; i >= 0; --i) {
            result[i] += ops;
            cnt += boxes[i] == '1' ? 1 : 0;
            ops += cnt;
        }
        return result;
    }
};
